src/init/ServerInitializer.ts

```diff
@@ -9,18 +10,21 @@ import { Initializer } from './Initializer';
  */
 export class ServerInitializer extends Initializer implements Finalizable {
   private readonly serverFactory: HttpServerFactory;
+  private readonly serverListener: ServerListener;
   private readonly port: number;
 
   private server?: Server;
 
-  public constructor(serverFactory: HttpServerFactory, port: number) {
+  public constructor(serverFactory: HttpServerFactory, serverListener: ServerListener, port: number) {
     super();
     this.serverFactory = serverFactory;
+    this.serverListener = serverListener;
     this.port = port;
   }
 
   public async handle(): Promise<void> {
     this.server = this.serverFactory.startServer(this.port);
+    await this.serverListener.handleSafe(this.server);
```

Comment:
Mmm `serverListener` is really a misnomer here; it is the object that attaches a server listener.
I would have called it an initializer, but then `ServerInitializer` is already a thing.

However, it seems that `ServerInitializer` is not doing a lot now. Could we just merge `ServerInitializer` into the concrete server factory? But then we have the `finalize` indeed…

Maybe `ServerInitializer` becomes `ServerManager` or `ServerRunner`, and `ServerListener` becomes `ServerInitializer`, because it really only executes once.

That said, given that it only executes once… maybe the (current) `ServerListener` (renamed to `ServerInitializer`) should instead be a parameter to the server factory; it's their job to construct them in. Then the (current) `ServerInitializer` indeed becomes simply `ServerRunner`, whose job it is to start a server on a certain port of socket.

Reply:
But `ServerRunner` still extends `Initializer` right? And `ServerInitializer` would not extend `Initializer`, so that might cause some confusion.

Reply:
Ah right, `Initializer` is a thing too. Even though it's just a simple async handler. Mmm… So do we really need it on `ServerRunner`?
Or any other naming suggestions? `ServerConfigurator` or so for the current listeners?

Reply:
> So do we really need it on `ServerRunner`?

We need an `Initializer` that triggers everything related to creating a server and listening to the port, which is what `ServerInitializer` currently is. An `Initializer` is a handler that gets executed when the process starts so it's a fixed concept in CSS. I'm OK with `ServerConfigurator`. Most names would be fine to me, would just not reuse the term `Initializer`.

Reply:
`ServerConfigurator` it is!

Reply:
> That said, given that it only executes once… maybe the (current) `ServerListener` (renamed to `ServerInitializer`) should instead be a parameter to the server factory; it's their job to construct them in.

Problem is that `startServer` is a sync function while the `ServerConfigurator`s are handlers and thus async. Although I don't really see a problem with making `startServer` async as well.